src/plugin.ts

```diff
@@ -0,0 +1,340 @@
+ *
+ * This reviver function is executed bottom up and is essentially
+ * creating a new object with a well known schema that the rest of the
+ * plugin can work with. It looks for certain fields that always appear in
+ * the guard results, but appear in different locations. It finds those fields
+ * and then pulls them up the object, dropping any fields that we don't
+ * care about. For example guard may return
+ *
+ * {
+ *   Clause: {
+ *     Unary: {
+ *       check: {
+ *         UnResolved: {
+ *           value: {
+ *             traversed_to: {...} // we only care about this!!!
+ *           }
+ *         }
+ *       }
+ *     }
+ *   }
+ * }
+ *
+ * Or it may return
+ *
+ * {
+ *   Rule: {
+ *     checks: [{
+ *       Block: {
+ *         unresolved: {
+ *           traversed_to: {...} // we only care about this!!!
+ *         }
+ *       }
+ *     }]
+ *   }
+ * }
+ *
+ * In the above example we only care about the 'traversed_to' field,
+ * so this reviver function will grab that field and pull it up the object, dropping
+ * the fields we don't care about, ending with something like
+ * {
+ *   checks: [{
+ *     resolved: false,
+ *     traversed: {...}
+ *   }]
+ * }
+ *
+ */
+function reviver(key: string, value: any): any {
+  if (value !== null && typeof value === 'object' && !Array.isArray(value)) {
+    return extractNestedObject(value);
+  } else if (key === 'checks' && Array.isArray(value)) {
+    return extractNestedChecks(value);
+  } else if (key === 'not_compliant') {
+    return Object.values(value).map((v: any) => v.Rule);
+  }
+  return value;
+}
+
+/**
+ * Extract a nested 'checks' object. This also handles checks
+ * nested within checks. It will grab the checks at the level below
+ * and pull it up to the next level.
+ */
+function extractNestedChecks(checks: any[]): any[] {
+  if (checks.some(check => Object.values(check).some((value: any) => {
+    if (typeof value === 'object') {
+      if (value.hasOwnProperty('checks')) {
+        return true;
+      }
+    }
+    return false;
+  }))) {
+    return checks.flatMap(check => {
+      return Object.values(check).flatMap((checkValue: any) => {
+        return Object.values(checkValue.checks).flatMap((nestedCheckValue: any) => {
+          return {
+            ...nestedCheckValue,
+            name: checkValue.name,
+            messages: checkValue.messages ?? nestedCheckValue.messages,
+          };
+        });
+      });
+    });
+  }
+  return checks.flatMap(check => {
+    if (Object.keys(check).includes('traversed')) {
+      return check;
+    }
+    return Object.values(check);
+  });
+}
+
+/**
+ * Extract a nested object and pull it up a level
+ */
+function extractNestedObject(object: any): any {
+  let newObject = object;
+  Object.entries(object).forEach(([level1NestedKey, level1NestedValue]) => {
+    const nestedValue = level1NestedValue as any;
+    switch (level1NestedKey.toLowerCase()) {
+      case 'unresolved':
```

Comment:
It would be helpful to have a small comment for each of these cases explaining what they mean.

Reply:
Ok added some comments. Let me know if those are good.